recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -148,15 +123,41 @@ def _configure_cmake(self):
         self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
+    def validate(self):
+        if self.options.with_ssl and not bool(self.options["mongo-c-driver"].with_ssl):
+            raise ConanInvalidConfiguration("mongo-cxx-driver with_ssl=True requires mongo-c-driver with a ssl implementation")
```

Comment:
No new code here, everything is from `configure()` or `build()` methods.

I followed the recommendation from here:
https://blog.conan.io/2020/12/14/New-conan-release-1-32.html
the `validate()` method should take over all responsibility for raising `ConanInvalidConfiguration` errors